beacon-chain/blockchain/receive_attestation.go

```diff
@@ -88,30 +93,48 @@ func (s *Service) spawnProcessAttestationsRoutine(stateFeed *event.Feed) {
 		}
 
 		st := slots.NewSlotTicker(s.genesisTime, params.BeaconConfig().SecondsPerSlot)
+		pat := slots.NewSlotTickerWithOffset(s.genesisTime, -reorgLateBlockCountAttestations, params.BeaconConfig().SecondsPerSlot)
 		for {
 			select {
 			case <-s.ctx.Done():
 				return
+			case <-pat.C():
+				root := s.UpdateHead(s.ctx)
```

Comment:
shouldnt this be gated by the flag

Reply:
It can, but we discussed this at length and this is something that lighthouse did for a while it's beneficial even without the reorg: we do most of the attestation processing early, so that at the beginning of the slot there's less work to do. I'm inclined to leave it ungated, but if you feel strongly about this we can gate it as well.